### PR TITLE
mongo-c-driver: add v1.26.2, bump deps, remove old versions

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

File: recipes/mongo-c-driver/all/conanfile.py [hunk 1]
```diff
@@ -153,6 +153,12 @@ def generate(self):
             # Should be added because of
             # https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-initonceexecuteonce
             tc.preprocessor_definitions["_WIN32_WINNT"] = "0x0600"
+        tc.cache_variables["BUILD_VERSION"] = self.version
+        # Skip some fragile checks
+        # https://github.com/mongodb/mongo-c-driver/blob/1.25.3/src/libmongoc/CMakeLists.txt#L266-L276
+        tc.variables["HAVE_ASN1_STRING_GET0_DATA"] = True  # Requires OpenSSL 1.1.0+
+        # https://github.com/mongodb/mongo-c-driver/blob/1.25.3/src/libmongoc/CMakeLists.txt#L366-L375
+        tc.variables["SASL2_HAVE_SASL_CLIENT_DONE"] = True # Requires Cyrus-SASL 2.1.23+
```
Comment (RubenRBS) [inline recipes/mongo-c-driver/all/conanfile.py:161]: Should this be conditional on the cyrus and openssl requirements being there?

**Review by RubenRBS — COMMENTED** [review]
Thanks! Minor question before merging

File: recipes/mongo-c-driver/all/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/mongo-c-driver/all/conanfile.py:161]: Kind of, but it only creates a CMake warning about an unused variable at most.

**Comment by Ahajha** [comment]
1.26.0 is also released, @valgur would you mind if I updated this PR to include that version as well? (Would resolve https://github.com/conan-io/conan-center-index/issues/20879)

**Comment by valgur** [comment]
@Ahajha Go ahead. Feel free to do so for any other PRs as well.

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Comment by valgur** [comment]
Thanks, @Ahajha!

**Review by danimtb — APPROVED** [review]

**Pull request merged**